Mono.Counters/NativeCounter.cs

```diff
@@ -34,9 +34,21 @@ internal object GetValue(byte[] array)
                     case CounterType.IntPtr:
                         return MemoryMarshal.Read<IntPtr>(array);
                     case CounterType.Long:
-                        return MemoryMarshal.Read<long>(array);
+                        long value = MemoryMarshal.Read<long>(array);
+                        if (Unit.Equals(CounterUnit.Time))
+                            return TimeSpan.FromTicks(value / 10000);
+                        return value;
                     case CounterType.ULong:
-                        return MemoryMarshal.Read<ulong>(array);
+                        ulong value2 = MemoryMarshal.Read<ulong>(array);
+                        if (Unit.Equals(CounterUnit.Time))
```

Comment:
Using Equals for enums is pretty icky, because it has the signature: `Equals(object other)`. That implicitly causes a boxing of the enum value so it allocates an object every time equals is called.

`==` is a completely safe operation for enums.

```suggestion
                        if (Unit == CounterUnit.Time)
```